cpp/include/cudf/quantiles.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.hpp"
+
+namespace cudf {
+
+/**
+ * @brief Interpolation method to use, when the desired quantile lies between 
+ * two data points i and j
+ * 
+ */
+enum quantile_method{
+  QUANTILE_LINEAR =0,      ///< Linear interpolation between i and j
```

Comment:
Cython doesn't yet support enum class https://github.com/cython/cython/issues/1603.

Reply:
This would muddy up the `cudf` namespace right? We'd now have `cudf::LINEAR`, `cudf::LOWER` etc. At least with the `QUANTILE_` prefix, others know not to use it. With enum class it would've been great because then they're in the `interpolation` namespace and it would have to be mentioned everywhere it's used.

Reply:
It wouldn't be `cudf::LINEAR`, it would be `cudf::interpolation::linear`.

Reply:
Looks like enum class is doable in Cython: https://stackoverflow.com/questions/31001918/wrap-enum-class-for-cython

Reply:
@harrism That's how @kovaltan tried to do it in https://github.com/rapidsai/cudf/pull/1005/commits/d03519470ed8f3f493ce3ef99b97bd71b98d51ba but he then reverted it the very next commit https://github.com/rapidsai/cudf/pull/1005/commits/1ac55a323fe4e664aba20b88ca29265245749207 because apparently it "fails at cython"